### PR TITLE
Define PACTA Params Schema

### LINKED ISSUE #62: Define input parameters and JSON schema.
Define JSON schema for input params

### LINKED ISSUE #66: Docker: Mount external FS under `/mnt`
Keep things tidy, and explicitly use `/mnt/foo` for external data (`/mnt/pacta-data`, `/mnt/portfolios`, etc.).

### LINKED ISSUE #67: Split Config and Params
Explcitly split Config (instance-level settings) from Params (run-level settings)

Example:
Data Dir (where pacta-data lives) is a config, since that will be the same for all instances run by an instance
Holdings date is a param, since that will change from portfolio to portfolio.

https://12factor.net/config

### PR DESCRIPTION
**Pull request by AlexAxthelm**
* Make use of the parameter inheritence from `pacta.workflow.utils:::parse_params()`.
  * Parameters to inherit are located as files in `inst/extdata/parameters` (accessible within R using `system.file(..., package = "workflow.pacta")`.
* Split system config out from portfolio parameters
  * Config is controlled via envvars, params are passed as JSON.
* Change mount points to container to `/mnt`

- [x] Depends on: https://github.com/RMI-PACTA/pacta.workflow.utils/pull/13
- [x] Depends on https://github.com/RMI-PACTA/pacta.portfolio.allocate/pull/49

Closes #62 
Closes #67 
Closes #66

### CODE REVIEW

**Comment by AlexAxthelm** [comment]
@cjyetman @jdhoffa marking this as ready for review, with the caveat that I'm probably going to do some re-organization of the top level keys of the JSON params, based on what I see coming with [`workflow.pacta.report`](https://github.com/RMI-PACTA/workflow.pacta.report/issues/12). We've previously discussed having the JSON along the lines of 
```json
{
  "portfolio_files": ["1.csv", "2.csv"],
  "analysis": {
     "start_year": 2023,
     "equity_market_list": ["blah blah blah"]
   },
   "reporting": {
     "portfolio_name": "Foo Portfolio",
     "investor_name": "Bar Investor"
   }
}
```

which is not implemented here, but that can probably come as a separate PR

File: DESCRIPTION [hunk 1]
```diff
@@ -27,16 +27,17 @@ Roxygen: list(markdown = TRUE)
...
     pacta.workflow.utils,
     readr
 Remotes:
-    RMI-PACTA/pacta.portfolio.allocate,
+    RMI-PACTA/pacta.portfolio.allocate@fix/48-remove-globalenv,
     RMI-PACTA/pacta.portfolio.audit,
     RMI-PACTA/pacta.portfolio.import,
     RMI-PACTA/pacta.portfolio.utils,
-    RMI-PACTA/pacta.workflow.utils
+    RMI-PACTA/pacta.workflow.utils@fix/integration-issues-with-worfklow.pacta
```
Comment (cjyetman) [inline DESCRIPTION]: do you want the branch names merged to main?

**Review by cjyetman — DISMISSED** [review]

File: DESCRIPTION (hunk 1 above)
Comment (AlexAxthelm) [reply DESCRIPTION]: I'd rather not. Those are the two PRs mentioned at the top.

File: DESCRIPTION [hunk 2]
```diff
@@ -39,4 +40,4 @@ Remotes:
...
```
Comment (jdhoffa) [inline DESCRIPTION]: I approved https://github.com/RMI-PACTA/pacta.workflow.utils/pull/13

**Review by jdhoffa — CHANGES_REQUESTED** [review]
Comment to revert changes to `DESCRIPTION` otherwise LGTM

File: DESCRIPTION (hunk 2 above)
Comment (AlexAxthelm) [reply DESCRIPTION]: reverted.

File: DESCRIPTION (hunk 1 above)
Comment (AlexAxthelm) [reply DESCRIPTION]: resolved.

**Review by jdhoffa — DISMISSED** [review]

**Review by jdhoffa — APPROVED** [review]

**Pull request merged**